iexec/src/main/scala/gridscale/iexec/package.scala

```diff
@@ -0,0 +1,107 @@
+package gridscale
+
+import effectaside._
+import gridscale.cluster.BatchScheduler.BatchJob
+import gridscale.cluster.{ BatchScheduler, HeadNode }
+import gridscale.tools._
+import monocle.macros._
+import squants._
+import squants.information._
+
+package object iexec {
+
+  case class IEXECJobDescription(
+    workDirectory: String,
+    IexecFilesPath: String,
+    dappAddress: String,
+    arguments: String,
+    dappCost: Int)
+
+  object impl {
+
+    def toIexec[S](server: S, description: IEXECJobDescription)(implicit hn: HeadNode[S], system: Effect[System]) = {
+
+      loginIexecAccount(server, description)
+      verifyIexecAccountBalance(server, description)
+
+      s"""
+         |#!/bin/bash
+         |export PATH="${description.IexecFilesPath}:$${PATH}"
+         |cd ${description.workDirectory}
+         |DEBUG='iexec:submit' iexec submit ${description.arguments} --dapp ${description.dappAddress}
+       """.stripMargin
+    }
+
+    def loginIexecAccount[S](server: S, description: IEXECJobDescription)(implicit hn: HeadNode[S], system: Effect[System]) = {
+      if (!new java.io.File(s"""/${description.workDirectory}/account.json""").exists) {
+        hn.execute(server, s"""cd ${description.workDirectory} && export PATH="${description.IexecFilesPath}:$${PATH}" && iexec account login""")
+      }
```

Comment:
Would be good to find out whether the token generated with `iexec acount login` expires at some point? Maybe ask the team on Slack